src/main/resources/public/js/components/rightPane.js

```diff
@@ -0,0 +1,258 @@
+        },
+        plusIng () {
+            this.displayIngDiv = true;
+        },
+        delRec () {
+            $.ajax({
+                type: 'DELETE',
+                url: `/recipes/${this.$store.state.currentRec.id}`,
+                dataType: 'json',
+            })
+                .fail(this.fail)
+                .done(this.reset);
+        },
+        fail (jqXHR) {
+            //      document.getElementById('fail').play();
+            let message = '';
+            if (jqXHR.readyState === 0) {
+                message = 'Failed to contact server.';
+            } else if (jqXHR.hasOwnProperty('responseJSON')) {
+                message = jqXHR.responseJSON.message;
+            } else if (jqXHR.hasOwnProperty('responseText')) {
+                message = jqXHR['responseText'];
+            } else {
+                message = "An unknown error has occured."
+            }
+            this.$emit('update-err', message);
+        },
+        applyRecRename () {
+            if (this.rename === this.$store.state.currentRec.name) {
+                this.cancelRecRename();
+                return;
+            }
+            this.$emit('update-err', '');
+            const otherRec = JSON.parse(JSON.stringify(this.$store.state.currentRec));
+            otherRec.name = this.rename;
+            $.post({
+                url: 'recipes',
+                data: JSON.stringify(otherRec),
+                contentType: 'application/json',
+                dataType: 'json',
+            }).done(
+                (data) => {
+                    this.$store.commit('updateCurRec', data);
+                    this.$store.commit('sortRecList');
+                },
+            ).fail(
+                (jqXHR) => {
+                    this.$store.state.showRecTitle = true;
+                    this.showRename = false;
+                    this.fail(jqXHR);
+                },
+            );
+            this.showRename = false;
+            this.showRecTitle = true;
+        },
+        reset () {
+            this.showContainer = false;
+            this.$store.commit('updateCurRec', {});
+            // On page load, gets the recipe list, and appends as buttons to a ul in left pane.
+            $.get('/recipes', 'json')
+                .done(
+                    (recipeList) => {
+                        this.$store.commit('updateRecList', recipeList);
+                        this.$store.commit('sortRecList');
+                    },
+                ).fail(this.fail);
+
+            this.$emit('update-err', '');
+        },
+        showIng (idx) {
+            if (idx === this.editIngIdx) {
+                return false;
+            }
+            return true;
+        },
+        success (recipe) {
+            console.log('success!');
+            this.$store.commit('setCurRec', recipe);
+        },
+        delIng (idx) {
+            this.$store.state.currentRec.ingredients.splice(idx, 1)
```

Comment:
This is actually against the rules. You aren't allowed to modify objects in the store directly; you're supposed to "commit" something. However, in this case, I don't think you want to modify currentRec at all. I think you should clone currentRec, remove the appropriate ingredient, and send the json-stringify of that.

Upon success, you should update the ingredients in currentRec to reflect the new reality.

That way, if the deletion actually failed, currentRec would reflect the recipe as it was before the ing-deletion was attempted.

Reply:
Oh whoops forgot
I will do that

Reply:
Done.